shared/chat/inbox/row/index.js

```diff
@@ -0,0 +1,76 @@
+// @flow
+import React from 'react'
+import BigTeamHeader from './big-team-header/container'
+import BigTeamChannel from './big-team-channel/container'
+import FilterBigTeamChannel from './filter-big-team-channel/container'
+import FilterSmallTeamChannel from './filter-small-team/container'
+import SmallTeam from './small-team/container'
+import * as Constants from '../../../constants/chat'
+
+type MakeRowOptions = {
+  channelname: string,
+  conversationIDKey: ?Constants.ConversationIDKey,
+  filtered: boolean,
+  isActiveRoute: boolean,
+  teamname: string,
+  type: 'small' | 'bigHeader' | 'bigTeamsLabel' | 'big' | 'divider',
+}
+const makeRow = (options: MakeRowOptions) => {
+  if (options.filtered) {
+    switch (options.type) {
+      case 'big':
+        return (
+          <FilterBigTeamChannel
+            key={`${options.teamname}:${options.channelname}`}
+            conversationIDKey={options.conversationIDKey}
+            channelname={options.channelname}
+            isActiveRoute={options.isActiveRoute}
+            teamname={options.teamname}
+          />
+        )
+      case 'small':
+        return (
+          <FilterSmallTeamChannel
+            key={options.conversationIDKey}
+            conversationIDKey={options.conversationIDKey}
+            channelname={options.channelname}
+            isActiveRoute={options.isActiveRoute}
+            teamname={options.teamname}
+          />
+        )
+    }
+  } else {
```

Comment:
perhaps handle the `bigHeader` case here explicitly and make the default fallthrough below throw an Error?

Reply:
there is no big header when you're filtered. i did make it throw an error on unknown type